src/pages/my_page/user_profile/ProfilePictureUploader.tsx

```diff
@@ -0,0 +1,147 @@
+import AddAPhotoIcon from '@mui/icons-material/AddAPhoto';
+import CancelIcon from '@mui/icons-material/Cancel';
+import { Box, Button, CircularProgress, Skeleton, Typography } from '@mui/material';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { fetchProfilePicture, uploadProfilePicture } from '../../../api/cristinApi';
+import { PageSpinner } from '../../../components/PageSpinner';
+import { setNotification } from '../../../redux/notificationSlice';
+import { dataTestId } from '../../../utils/dataTestIds';
+
+interface ProfilePictureUploaderProps {
+  id: string;
+}
+
+export const ProfilePictureUploader = ({ id }: ProfilePictureUploaderProps) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const profilePictureQuery = useQuery({
+    queryKey: ['picture', id],
+    queryFn: () => fetchProfilePicture(id),
+    meta: { errorMessage: false },
+    retry: false,
+  });
+
+  const mutateProfilePicture = useMutation({
+    mutationFn: (base64String: string) => uploadProfilePicture(id, base64String),
+    onSuccess: () => {
+      dispatch(setNotification({ message: t('feedback.success.upload_profile_photo'), variant: 'success' }));
+      setIsLoading(false);
+      profilePictureQuery.refetch();
```

Comment:
Ser du har satt returtypen til `uploadProfilePicture` til å inneholde `base64Data`. Kan ikke den brukes igjen her, for å unngå enda ett kall?
Etter en kjapp sjekk kan det se ut som at man da må bruke `useQueryClient` som vi gjorde i går, så kanskje er det ikke noen stor gevinst, men verdt å sjekke ut